mlflow/store/dbfs_artifact_repo.py

```diff
@@ -69,14 +71,8 @@ def log_artifact(self, local_file, artifact_path=None):
         else:
             http_endpoint = self._get_dbfs_endpoint(os.path.basename(local_file))
         with open(local_file, 'rb') as f:
-            response = self._databricks_api_request(
+            self._databricks_api_request(
                 endpoint=http_endpoint, method='POST', data=f, allow_redirects=False)
-            if response.status_code == 409:
-                raise MlflowException('File already exists at {} and can\'t be overwritten.'
```

Comment:
Did we check what the error from the server is in this case?

Reply:
Good question, we end up logging it, though it's admittedly kind of ugly:

```
Traceback (most recent call last):
  File "/Users/siddharthmurching/code/mlflow/mlflow/utils/rest_utils.py", line 66, in http_request_safe
    js_dict = json.loads(response.text)
  File "/Users/siddharthmurching/miniconda3/envs/mlflow_python3/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/Users/siddharthmurching/miniconda3/envs/mlflow_python3/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Users/siddharthmurching/miniconda3/envs/mlflow_python3/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "examples/quickstart/mlflow_tracking.py", line 21, in <module>
    log_artifacts("outputs")
  File "/Users/siddharthmurching/code/mlflow/mlflow/tracking/fluent.py", line 223, in log_artifacts
    MlflowClient().log_artifacts(run_id, local_dir, artifact_path)
  File "/Users/siddharthmurching/code/mlflow/mlflow/tracking/client.py", line 166, in log_artifacts
    artifact_repo.log_artifacts(local_dir, artifact_path)
  File "/Users/siddharthmurching/code/mlflow/mlflow/store/dbfs_artifact_repo.py", line 91, in log_artifacts
    endpoint=endpoint, method='POST', data=f, allow_redirects=False)
  File "/Users/siddharthmurching/code/mlflow/mlflow/store/dbfs_artifact_repo.py", line 33, in _databricks_api_request
    return http_request_safe(host_creds=host_creds, endpoint=endpoint, **kwargs)
  File "/Users/siddharthmurching/code/mlflow/mlflow/utils/rest_utils.py", line 68, in http_request_safe
    raise MlflowException("%s. Response body: '%s'" % (base_msg, response.text))
mlflow.exceptions.MlflowException: API request to endpoint /dbfs/databricks/mlflow/0/038ce380607e4f26a6e86ccc9a5e84be/artifacts/test.txt failed with error code 409 != 200. Response body: '<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 409 </title>
</head>
<body>
<h2>HTTP ERROR: 409</h2>
<p>Problem accessing /dbfs/databricks/mlflow/0/038ce380607e4f26a6e86ccc9a5e84be/artifacts/test.txt. Reason:
<pre>    File already exists, cannot overwrite: &apos;/databricks/mlflow/0/038ce380607e4f26a6e86ccc9a5e84be/artifacts/test.txt&apos;</pre></p>
<hr />
</body>
</html>
'
```

Reply:
Edit: pushed an update that hides the JSONDecodeError part of the exception

```
Running mlflow_tracking.py
Traceback (most recent call last):
  File "examples/quickstart/mlflow_tracking.py", line 21, in <module>
    log_artifacts("outputs")
  File "/Users/siddharthmurching/code/mlflow/mlflow/tracking/fluent.py", line 223, in log_artifacts
    MlflowClient().log_artifacts(run_id, local_dir, artifact_path)
  File "/Users/siddharthmurching/code/mlflow/mlflow/tracking/client.py", line 166, in log_artifacts
    artifact_repo.log_artifacts(local_dir, artifact_path)
  File "/Users/siddharthmurching/code/mlflow/mlflow/store/dbfs_artifact_repo.py", line 91, in log_artifacts
    endpoint=endpoint, method='POST', data=f, allow_redirects=False)
  File "/Users/siddharthmurching/code/mlflow/mlflow/store/dbfs_artifact_repo.py", line 33, in _databricks_api_request
    return http_request_safe(host_creds=host_creds, endpoint=endpoint, **kwargs)
  File "/Users/siddharthmurching/code/mlflow/mlflow/utils/rest_utils.py", line 74, in http_request_safe
    raise MlflowException("%s. Response body: '%s'" % (base_msg, response.text))
mlflow.exceptions.MlflowException: API request to endpoint /dbfs/databricks/mlflow/0/ae3c099bc3cf44f696102d44c615fcc1/artifacts/test.txt failed with error code 409 != 200. Response body: '<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 409 </title>
</head>
<body>
<h2>HTTP ERROR: 409</h2>
<p>Problem accessing /dbfs/databricks/mlflow/0/ae3c099bc3cf44f696102d44c615fcc1/artifacts/test.txt. Reason:
<pre>    File already exists, cannot overwrite: &apos;/databricks/mlflow/0/ae3c099bc3cf44f696102d44c615fcc1/artifacts/test.txt&apos;</pre></p>
<hr />
</body>
</html>
'
```

Reply:
Yeah I agree with you here.